packages/flutter/lib/src/widgets/dismissable.dart

```diff
@@ -249,31 +263,51 @@ class _DismissableState extends State<Dismissable> {
   }
 
   Widget build(BuildContext context) {
+    Widget background;
+    if (config.backgrounds != null)
+      background = config.backgrounds[_dismissDirection] ?? config.backgrounds[config.direction];
+
     if (_resizeAnimation != null) {
       // we've been dragged aside, and are now resizing.
       assert(() {
         if (_resizeAnimation.status != AnimationStatus.forward) {
           assert(_resizeAnimation.status == AnimationStatus.completed);
           throw new WidgetError(
-            'Dismissable widget completed its resize animation without being removed from the tree.\n'
-            'Make sure to implement the onDismissed handler and to immediately remove the Dismissable '
+            'A dismissed Dismissable widget is still part of the tree.\n' +
+            (config.key != null ? '' : "Dismissable widgets must specifiy a key but this widget's key is null.\n") +
```

Comment:
Ok. We should make the message say that then. "If a Dismissable has another Dismissable sibling, then it must specify a key. This Dismissable does not have a key." or something. (Avoid "but" unless you actively go and check if it has such a sibling. I think you could do that relatively easily if you want to, just get the parent, then walk the children and count Dismissables. But it's more complicated than that because what really matters is sibling subtrees that happen to have Dismissables in the same place, not just the Dismissables themselves being siblings.)